libs/prelude/Prelude/Nat.idr

```diff
@@ -386,16 +393,15 @@ log2 (S n) = log2NZ (S n) SIsNotZ
 --------------------------------------------------------------------------------
 -- GCD and LCM
 --------------------------------------------------------------------------------
-partial
-gcd : Nat -> Nat -> Nat
+gcd : (a: Nat) -> (b: Nat) -> {auto ok: NotBothZero a b} -> Nat
 gcd a Z = a
-gcd a b = assert_total (gcd b (a `modNat` b))
+gcd Z b = b
+gcd a (S b) = assert_total $ gcd (S b) (modNatNZ a (S b) SIsNotZ)
```

Comment:
Would it be possible to investigate the use of `assert_smaller` instead of `assert_total` here? I think it would be nicer to have in Prelude at least, but it's not essential.